src/metabase/lib/metric.cljc

```diff
@@ -41,19 +57,57 @@
         (lib.metadata.calculation/type-of query stage-number metric-metadata))
       :type/*))
 
+(defn- fallback-display-name []
+  (i18n/tru "[Unknown Metric]"))
+
 (defmethod lib.metadata.calculation/display-name-method :metadata/metric
   [_query _stage-number metric-metadata _style]
   (or ((some-fn :display-name :name) metric-metadata)
-      (i18n/tru "[Unknown Metric]")))
+      (fallback-display-name)))
 
 (defmethod lib.metadata.calculation/display-name-method :metric
   [query stage-number [_tag _opts metric-id-or-name] style]
   (or (when-let [metric-metadata (resolve-metric query metric-id-or-name)]
         (lib.metadata.calculation/display-name query stage-number metric-metadata style))
-      (i18n/tru "[Unknown Metric]")))
+      (fallback-display-name)))
+
+(mu/defn ^:private aggregating-by-metric? :- :boolean
+  "Whether a given stage of a query currently includes a `:metric` ref clause in its aggregations."
+  [query        :- ::lib.schema/query
+   stage-number :- :int
+   metric-id    :- ::lib.schema.id/metric]
+  (let [{aggregations :aggregation} (lib.util/query-stage query stage-number)]
+    (boolean
+     (some (fn [[tag :as clause]]
+             (when (= tag :metric)
+               (let [[_tag _opts id] clause]
+                 (= id metric-id))))
+           aggregations))))
+
+(defmethod lib.metadata.calculation/display-info-method :metadata/metric
+  [query stage-number {:keys [id description], :as metric-metadata}]
+  (assoc ((get-method lib.metadata.calculation/display-info-method :default) query stage-number metric-metadata)
+         :description description
+         :selected    (aggregating-by-metric? query stage-number id)))
```

Comment:
We usually omit `:selected` unless it's `true`, I think.

Reply:
Fixed 🔧